### PR TITLE
Fix C# Json parsing of well-known type Boolvalue

### PR DESCRIPTION
**Pull request by jhickson**
A JSON string generated by the JsonFormatter  for a BoolValue could not be in turn parsed by JsonParser because the JsonParser did not recognise BoolValue as a well-known type and so expected a JSON object.

### CODE REVIEW

**Comment by jhickson** [comment]
I signed it!

**Comment by jhickson** [comment]
This is a fix for #1509.

File: csharp/src/Google.Protobuf.Test/WellKnownTypes/WrappersTest.cs [hunk 1]
```diff
@@ -417,5 +417,12 @@ public void ClearWithReflection()
             TestWellKnownTypes.Descriptor.Fields[TestWellKnownTypes.StringFieldFieldNumber].Accessor.Clear(message);
             Assert.IsNull(message.StringField);
         }
+
+        [Test]
+        public void GivenBoolValueWhenPerformingRoundTripEncodingViaJsonThenShouldNotExpectObjectAtTopLevel()
+        {
+            var value = new BoolValue { Value = true };
+            Assert.That(value, Is.EqualTo(JsonParser.Default.Parse<BoolValue>(JsonFormatter.Default.Format(value))));
```
Comment (jskeet) [inline csharp/src/Google.Protobuf.Test/WellKnownTypes/WrappersTest.cs]: Could you rewrite this to use Assert.AreEqual, to be consistent with the rest of the tests?

**Comment by jskeet** [comment]
Actual change looks good - if you could make the requested test change and squash it back into one commit, I'll merge.

File: csharp/src/Google.Protobuf.Test/WellKnownTypes/WrappersTest.cs (hunk 1 above)
Comment (jhickson) [inline csharp/src/Google.Protobuf.Test/WellKnownTypes/WrappersTest.cs]: Will do - sorry I've become used to using the fluent approach.

**Comment by jhickson** [comment]
Requested test change committed and pushed.

**Comment by jskeet** [comment]
Great, thanks - will review and merge shortly, assuming it goes green.

**Comment by jskeet** [comment]
Have also just noticed the name and placement of the test - I'll send a separate PR after merging to move that to JsonParser or JsonFormatter, where I suspect we've already got similar tests.

**Comment by jhickson** [comment]
I see now: there a JsonParserTests.cs. I scrolled down as far as the WellKnownTypes folder and no further.  Is there a naming convention specified for tests anywhere?  I'd like to contribute more (rather than just flagging up issues I've found) but that's obviously going to be counter-productive if I keep getting the naming wrong!

**Comment by jskeet** [comment]
No very specific naming convention for tests, but typically something like `MethodUnderTest`, `MethodUnderTest_Situation` or `MethodUnderTest_Situation_Result`.

**Comment by jskeet** [comment]
(But as I mentioned, don't worry about it for this one - I'll move it and may add more tests anyway :)

**Pull request merged**